pkgs/build-support/trivial-builders/default.nix

```diff
@@ -306,7 +349,8 @@ rec {
         in
         if checkPhase == null then ''
           runHook preCheck
-          ${stdenv.shellDryRun} "$target"
+          # This is `shellDryRun` but for our particular `bash`.
+          ${bash} -n -O extglob "$target"
```

Comment:
`stdenv.shell` (or `stdenv.shellDryRun`) and `pkgs.runtimeShell` are not the same, the former is for the build platform, while the latter is for the host platform. This means that replacing both with the same version breaks cross compilation, and I don't see an obvious way to get around this. That's what I meant with splicing shenanigans being necessary in [this comment](https://github.com/NixOS/nixpkgs/pull/280592#discussion_r1468297591).

I don't know an obvious way to make this work in a clean way, so I'd say let's have that in a separate PR.